### PR TITLE
[ie/mediaklikk] Fix extractor

### LINKED ISSUE #11061: [MediaKlikk] Failed to download m3u8 information: HTTP Error 404: Not Found
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [x] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Worldwide (some contents only from Hungary)

### Provide a description that is worded well enough to be understood

I am not a programmer, I am a user of the yt-dlp.exe.
I want to download videos from the Mediaklikk.hu site.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://mediaklikk.hu/filmajanlo/cikk/az-ajto/']
[debug] Encodings: locale cp1250, fs utf-8, pref cp1250, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-109474-gc94988a781-20230103 (setts), ffprobe N-109474-gc94988a781-20230103, rtmpdump 2.3
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1830 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
[MediaKlikk] Extracting URL: https://mediaklikk.hu/filmajanlo/cikk/az-ajto/
[MediaKlikk] az-ajto: Downloading webpage
[MediaKlikk] 668177: Downloading webpage
[MediaKlikk] 668177: Downloading m3u8 information
WARNING: [MediaKlikk] Failed to download m3u8 information: HTTP Error 404: Not Found
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 668177: Downloading 1 format(s): rtmp
[debug] Invoking rtmp downloader on "rtmp://c201-node62-cdn.connectmedia.hu/4511/ca0ad2f957f4fe9d707e069e29ad6a41/66f0ade0/2016/01/25/13-28690-0000-1-/index.m3u8"
[download] Destination: Az ajtó [668177].m3u8
[debug] rtmpdump command line: rtmpdump --verbose -r rtmp://c201-node62-cdn.connectmedia.hu/4511/ca0ad2f957f4fe9d707e069e29ad6a41/66f0ade0/2016/01/25/13-28690-0000-1-/index.m3u8 -o "Az ajtó [668177].m3u8.part" --resume --skip 1
[rtmpdump] RTMPDump v2.3
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 0
[rtmpdump] DEBUG: Parsed host    : c201-node62-cdn.connectmedia.hu
[rtmpdump] DEBUG: Parsed app     : 4511/ca0ad2f957f4fe9d707e069e29ad6a41
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMP
[rtmpdump] DEBUG: Hostname : c201-node62-cdn.connectmedia.hu
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : 66f0ade0/2016/01/25/13-28690-0000-1-/index.m3u8
[rtmpdump] DEBUG: tcUrl    : rtmp://c201-node62-cdn.connectmedia.hu:1935/4511/ca0ad2f957f4fe9d707e069e29ad6a41
[rtmpdump] DEBUG: app      : 4511/ca0ad2f957f4fe9d707e069e29ad6a41
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.
[rtmpdump] Downloaded 0 bytes
[rtmpdump] RTMPDump v2.3
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 0
[rtmpdump] DEBUG: Parsed host    : c201-node62-cdn.connectmedia.hu
[rtmpdump] DEBUG: Parsed app     : 4511/ca0ad2f957f4fe9d707e069e29ad6a41
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMP
[rtmpdump] DEBUG: Hostname : c201-node62-cdn.connectmedia.hu
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : 66f0ade0/2016/01/25/13-28690-0000-1-/index.m3u8
[rtmpdump] DEBUG: tcUrl    : rtmp://c201-node62-cdn.connectmedia.hu:1935/4511/ca0ad2f957f4fe9d707e069e29ad6a41
[rtmpdump] DEBUG: app      : 4511/ca0ad2f957f4fe9d707e069e29ad6a41
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.


ERROR: rtmpdump exited with code 1
  File "yt_dlp\__main__.py", line 17, in <module>
  File "yt_dlp\__init__.py", line 1081, in main
  File "yt_dlp\__init__.py", line 1071, in _real_main
  File "yt_dlp\YoutubeDL.py", line 3607, in download
  File "yt_dlp\YoutubeDL.py", line 3582, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1615, in extract_info
  File "yt_dlp\YoutubeDL.py", line 1626, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1782, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1841, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 3015, in process_video_result
  File "yt_dlp\YoutubeDL.py", line 179, in wrapper
  File "yt_dlp\YoutubeDL.py", line 3483, in process_info
  File "yt_dlp\YoutubeDL.py", line 3203, in dl
  File "yt_dlp\downloader\common.py", line 466, in download
  File "yt_dlp\downloader\rtmp.py", line 212, in real_download
  File "yt_dlp\YoutubeDL.py", line 1092, in report_error
  File "yt_dlp\YoutubeDL.py", line 1020, in trouble


#################################################################
# nightly build output:
#################################################################
[debug] Command-line config: ['-vU', 'https://mediaklikk.hu/filmajanlo/cikk/az-ajto/']
[debug] Encodings: locale cp1250, fs utf-8, pref cp1250, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.09.17.232940 from yt-dlp/yt-dlp-nightly-builds [4a9bc8c36] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-109474-gc94988a781-20230103 (setts), ffprobe N-109474-gc94988a781-20230103, rtmpdump 2.3
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.3, websockets-13.0.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1835 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.09.17.232940 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.09.17.232940 from yt-dlp/yt-dlp-nightly-builds)
[MediaKlikk] Extracting URL: https://mediaklikk.hu/filmajanlo/cikk/az-ajto/
[MediaKlikk] az-ajto: Downloading webpage
[MediaKlikk] 668177: Downloading webpage
[MediaKlikk] 668177: Downloading m3u8 information
WARNING: [MediaKlikk] Failed to download m3u8 information: HTTP Error 404: Not Found
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 668177: Downloading 1 format(s): rtmp
[debug] Invoking rtmp downloader on "rtmp://c201-node61-cdn.connectmedia.hu/4511/72006b3c75306df8c6a96048e8b10d09/66f0aeab/2016/01/25/13-28690-0000-1-/index.m3u8"
[download] Destination: Az ajtó [668177].m3u8
[debug] rtmpdump command line: rtmpdump --verbose -r rtmp://c201-node61-cdn.connectmedia.hu/4511/72006b3c75306df8c6a96048e8b10d09/66f0aeab/2016/01/25/13-28690-0000-1-/index.m3u8 -o "Az ajtó [668177].m3u8.part" --resume --skip 1
[rtmpdump] RTMPDump v2.3
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 0
[rtmpdump] DEBUG: Parsed host    : c201-node61-cdn.connectmedia.hu
[rtmpdump] DEBUG: Parsed app     : 4511/72006b3c75306df8c6a96048e8b10d09
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMP
[rtmpdump] DEBUG: Hostname : c201-node61-cdn.connectmedia.hu
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : 66f0aeab/2016/01/25/13-28690-0000-1-/index.m3u8
[rtmpdump] DEBUG: tcUrl    : rtmp://c201-node61-cdn.connectmedia.hu:1935/4511/72006b3c75306df8c6a96048e8b10d09
[rtmpdump] DEBUG: app      : 4511/72006b3c75306df8c6a96048e8b10d09
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.
[rtmpdump] Downloaded 0 bytes
[rtmpdump] RTMPDump v2.3
[rtmpdump] (c) 2010 Andrej Stepanchuk, Howard Chu, The Flvstreamer Team; license: GPL
[rtmpdump] DEBUG: Parsing...
[rtmpdump] DEBUG: Parsed protocol: 0
[rtmpdump] DEBUG: Parsed host    : c201-node61-cdn.connectmedia.hu
[rtmpdump] DEBUG: Parsed app     : 4511/72006b3c75306df8c6a96048e8b10d09
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Number of skipped key frames for resume: 1
[rtmpdump] DEBUG: Protocol : RTMP
[rtmpdump] DEBUG: Hostname : c201-node61-cdn.connectmedia.hu
[rtmpdump] DEBUG: Port     : 1935
[rtmpdump] DEBUG: Playpath : 66f0aeab/2016/01/25/13-28690-0000-1-/index.m3u8
[rtmpdump] DEBUG: tcUrl    : rtmp://c201-node61-cdn.connectmedia.hu:1935/4511/72006b3c75306df8c6a96048e8b10d09
[rtmpdump] DEBUG: app      : 4511/72006b3c75306df8c6a96048e8b10d09
[rtmpdump] DEBUG: live     : no
[rtmpdump] DEBUG: timeout  : 30 sec
[rtmpdump] DEBUG: Failed to get last keyframe.
[rtmpdump] DEBUG: Closing connection.


ERROR: rtmpdump exited with code 1
  File "yt_dlp\__main__.py", line 17, in <module>
  File "yt_dlp\__init__.py", line 1081, in main
  File "yt_dlp\__init__.py", line 1071, in _real_main
  File "yt_dlp\YoutubeDL.py", line 3607, in download
  File "yt_dlp\YoutubeDL.py", line 3582, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1615, in extract_info
  File "yt_dlp\YoutubeDL.py", line 1626, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1782, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1841, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 3015, in process_video_result
  File "yt_dlp\YoutubeDL.py", line 179, in wrapper
  File "yt_dlp\YoutubeDL.py", line 3483, in process_info
  File "yt_dlp\YoutubeDL.py", line 3203, in dl
  File "yt_dlp\downloader\common.py", line 466, in download
  File "yt_dlp\downloader\rtmp.py", line 212, in real_download
  File "yt_dlp\YoutubeDL.py", line 1092, in report_error
  File "yt_dlp\YoutubeDL.py", line 1020, in trouble
```

### PR DESCRIPTION
**Pull request by szantnerb**
### Description of your *pull request* and other information

The m3u8 manifest filename was changed from `playlist.m3u8` to `index.m3u8`.

In this PR I needed to modify the `_extract_wowza_formats` method in the `InfoExtractor` class to be able to customize this parameter - but since I am not usually programming in python this might not be an optimal solution.

Fixes #11061 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/mediaklikk.py [hunk 1]
```diff
@@ -144,7 +144,10 @@ def _real_extract(self, url):
             raise ExtractorError('Unable to extract playlist url')
 
         formats = self._extract_wowza_formats(
-            playlist_url, video_id, skip_protocols=['f4m', 'smil', 'dash'])
+            playlist_url,
+            video_id,
+            m3u8_custom_manifest_filename='index.m3u8',
+            skip_protocols=['f4m', 'smil', 'dash'])
```
Comment (seproDev) [inline yt_dlp/extractor/mediaklikk.py:150]: As far as I can tell, the site no longer uses wowza. So let's just inline:
```py
'formats': self._extract_m3u8_formats(playlist_url, video_id),
```
Realistically, this whole extractor needs to be rewritten, but you don't need to do that as part of this.

**Review by seproDev — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/mediaklikk.py (hunk 1 above)
Comment (szantnerb) [reply yt_dlp/extractor/mediaklikk.py]: Thanks for your guidance - this makes more sense.

**Review by seproDev — APPROVED** [review]

File: yt_dlp/extractor/mediaklikk.py [hunk 2]
```diff
@@ -143,8 +143,7 @@ def _real_extract(self, url):
         if not playlist_url:
             raise ExtractorError('Unable to extract playlist url')
 
-        formats = self._extract_wowza_formats(
-            playlist_url, video_id, skip_protocols=['f4m', 'smil', 'dash'])
+        formats = self._extract_m3u8_formats(playlist_url, video_id)
```
Comment (bashonly) [inline yt_dlp/extractor/mediaklikk.py:146]: we could do this pre-emptively
```suggestion
        formats, subtitles = self._extract_m3u8_formats_and_subtitles(playlist_url, video_id)
```
and then make sure to return `subtitles` below

**Review by bashonly — CHANGES_REQUESTED** [review]
none of the tests work. could you add a working test, and mark the broken tests with `skip`? you could use the url from the linked issue

File: yt_dlp/extractor/mediaklikk.py (hunk 2 above)
Comment (szantnerb) [reply yt_dlp/extractor/mediaklikk.py]: Extracted and returned subtitles as well.

**Comment by szantnerb** [comment]
> none of the tests work. could you add a working test, and mark the broken tests with `skip`? you could use the url from the linked issue

the old tests are now skipped (unfortunately most of these videos are only served for a couple of days/weeks?), and I added a new test case with the video from the linked issue

File: yt_dlp/extractor/mediaklikk.py [hunk 3]
```diff
@@ -16,6 +16,16 @@ class MediaKlikkIE(InfoExtractor):
                         (?P<id>[^/#?_]+)'''
 
     _TESTS = [{
+        'url': 'https://mediaklikk.hu/filmajanlo/cikk/az-ajto/',
+        'info_dict': {
+            'id': '668177',
+            'title': 'Az ajtó',
+            'display_id': 'az-ajto',
+            'ext': 'mp4',
+            'upload_date': None,
```
Comment (bashonly) [inline yt_dlp/extractor/mediaklikk.py:25]:
```suggestion
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**